src/applications/caregivers/components/FormFields/FacilitySearch.jsx

```diff
@@ -15,11 +15,23 @@ const FacilitySearch = props => {
   const [error, setError] = useState(null);
   const [facilities, setFacilities] = useState([]);
 
-  const listProps = useMemo(() => ({ ...props, facilities, query }), [
-    facilities,
-    props,
-    query,
-  ]);
+  const facilityListProps = useMemo(
+    () => {
+      const { onChange, ...restOfProps } = props;
+      const setSelectedFacilities = facilityId => {
+        const facility = facilities.find(f => f.id === facilityId);
+        onChange({ veteranSelected: facility });
```

Comment:
This calls the onChange with our facility object to store it. Previously we were just storing the id. I anticipate more changes here and probably pulling it out to a method with future cards around the parent/child facility behavior.

Reply:
🤔 Do we have standards on how much data we want to make the client handle? This would expand the footprint from a few bytes to ~4k (for the sample response facility in the Lighthouse docs). I'd expect that's small enough to not matter, but folks on metered data may have constraints we should consider. 

That said, it's just 4k?

Reply:
No idea on standards for how much data to store. If we need to we could limit it to the name/address/id data (ie whatever we want to render on the confirmation screen). For now I just added the whole object. Good call though to be aware of performance concerns here.